### PR TITLE
Implement rr record --num-cores=N

### PR DESCRIPTION
**Pull request by hotsphink**
Fixes issue #1738.

While it seems good for --chaos to randomize the number of cores, there are times when I would like to control the core count separately from scheduler randomization. This implementation allows passing --num-cores=N with or without --chaos, and will override chaos mode's setting if both are given (but scheduling will still be randomized, it will just use the given number of cores.)

It also slightly helps resolve the confusion that people have about rr reporting 1 core by default. Though I could have been more explicit about that in the usage message, I suppose. ("default is 1 core, or a random number from 1-8 if chaos mode is enabled"?)

### CODE REVIEW

File: src/Scheduler.cc [hunk 1]
```diff
@@ -157,10 +157,16 @@ void Scheduler::set_enable_chaos(bool enable_chaos) {
    * return 1 to maximize throughput (since effectively we really only have
    * one core).
    */
+  ASSERT(current_, pretend_num_cores_ == 1);
```
Comment (rocallahan) [inline src/Scheduler.cc]: I don't think we need this assertion. Why did you add it?

Comment (hotsphink) [reply src/Scheduler.cc]: The idea was that --num-cores wins out over --chaos for setting the number of cores, and that is implemented by calling set_chaos_mode first then set_num_cores (and overwriting any effect set_chaos_mode may have had) -- which means that there's an implicit ordering dependency between the two calls.

You can remove it, or I could add a comment, or merge the two calls together to remove the ordering dependency entirely (set_chaos_and_cores?). Let me know what you'd prefer.

Comment (hotsphink) [reply src/Scheduler.cc]: Oh, sorry. Further evidence that I under-commented: this assert is how I am verifying that set_num_cores was not called before set_chaos_mode.

File: src/RecordCommand.cc [hunk 2]
```diff
@@ -391,6 +406,8 @@ static void setup_session_from_flags(RecordSession& session,
   session.scheduler().set_max_ticks(flags.max_ticks);
   session.scheduler().set_always_switch(flags.always_switch);
   session.set_enable_chaos(flags.chaos);
+  if (flags.num_cores)
```
Comment (rocallahan) [inline src/RecordCommand.cc]: Needs {} around the if body

File: src/Scheduler.cc (hunk 1 above)
Comment (rocallahan) [reply src/Scheduler.cc]: I think implementing that restriction using an assert here creates a dependency in `Scheduler` on how `RecordCommand` is supposed to work, which isn't great. So please remove the assert (and fix the code formatting issue I just mentioned). Also, add a comment to `setup_session_from_flags` explaining that `num_cores` has to be set after `enable_chaos`.

**Comment by rocallahan** [comment]
Thanks!
